pkg/controller/integration/integration_controller.go

```diff
@@ -55,6 +57,16 @@ import (
 )
 
 func Add(mgr manager.Manager, c client.Client) error {
+	err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Pod{}, "status.phase",
```

Comment:
I've checked and the context is used to cancel wait on informer(s) synchronisation. It'll probably work 99% of the time, but for correctness, it may be better to use `ctrl.SetupSignalHandler()` or find a way to pass the parent context.

Reply:
Yep, that was a question I had in mind, maybe  the `Add` method should always require a context ?

Reply:
That sounds good, it'll offer the ability to do some logic requiring a context if needed when adding a controller.

Reply:
done